### ISSUE TITLE
protected category shows 500 error 

### ISSUE THREAD

**Issue by user2024september** [Kunena/Kunena-Forum#9675]
**Describe the bug**
I made a new top level category
Access Control Type:User Groups
Set primary user group to Registered

**To Reproduce**
You would need sign in info to reproduce.
I would need a secure area to give it to you where other readers in this github dont see it only you?? Please provide it?

**Expected behavior**
When a registered user is given an URL to the new forum category and goes to the site and is not logged in 
they should get login message and form 
**Actual result**
When a registered user is given an URL to the new forum category and goes to the site and is not logged in they get a 500 error
they should get login message and form 
500 Internal Server Error :You may not be able to visit this page because of:

    an out-of-date bookmark/favourite
    a search engine that has an out-of-date listing for this site
    a mistyped address
    you have no access to this page
    The requested resource was not found.
    An error has occurred while processing your request.

If difficulties persist, please contact the website administrator and report the error below.

**Screenshots**

**System information (please complete the following information)**

Joomla version: 4
Kunena version: 6.02
Php version:8.2
Database version: 5.0

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] win 11
 - Browser [e.g. chrome, safari] firefox
 - Version [e.g. 22]22

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6] iPhone6
 - OS: [e.g. iOS8.1]iOS8.1
 - Browser [e.g. stock browser, safari]  safari
 - Version [e.g. 22]22

**Additional context**
I assume the 500 error is triggered because of the number 4 reason "you have no access to this page"

However when you are logged in there is no error and you correctly can see and use the forum

Expected behaviour: non logged user is shown the login form with message to login
Actual behaviour: non logged user is shown the 500 error which is unexpected behaviour

**This  looks like the cause** setting under security tab title "Registered Users Only" Set to Yes to limit all forum access to registered users (view and post).
Set to No to allow any Guest (non registered visitor) to at least see the forum. See 'Allow Guests to post/write' for additional Guest access control.

If I set to "no" then there is no 500 error for logged in or out going to forum both receive correct login message
If logged out go to protected category directly from link they get correct message without 500 error message.
If logged in go to protected category directly from link they get access to protected category correctly without 500 error message.
This is now working as expected

The problem remaining is that if If logged out go to forum or any category, whether the category is protected or not, they get the following
Please Log In
Only community members can participate in this forum. You must register or log in to contribute.

I want the main forum home page to be viewable by logged out showing to the public logged out viewers all the categories
and showing any category I set and want to be viewed by logged out
In other words I want Kunena to work correctly and show the categories I want to show to logged out and not shoow categories I have protected and do not want to show to   logged out with the correct message to log in to view.

If I set this to "yes" then the 500 error incorrectly shows to logged out visitors.
Please advise on how to correct this?

Dont know if this is related but I also get this system error message now?
Please check your Joomla menus: Default Menu Item in Entry Page is not pointing to Kunena.
